pkg/test/inmemoryresolver.go

```diff
@@ -12,33 +12,42 @@ import (
 	"github.com/Iotic-Labs/iotics-identity-go/pkg/register"
 )
 
+var (
+	CountDiscover int
+	CountRegister int
+)
+
 type InMemoryResolver struct {
-	documents map[string]*register.RegisterDocument
-	lock      *sync.RWMutex
+	documents     map[string]*register.RegisterDocument
+	lock          *sync.RWMutex
 }
 
 func NewInMemoryResolver(docs ...*register.RegisterDocument) *InMemoryResolver {
 	documents := map[string]*register.RegisterDocument{}
 	for _, v := range docs {
 		documents[v.ID] = v
 	}
+	CountDiscover = 0
+	CountRegister = 0
 	return &InMemoryResolver{
-		documents: documents,
-		lock:      &sync.RWMutex{},
+		documents:     documents,
+		lock:          &sync.RWMutex{},
 	}
 }
 
-func (c InMemoryResolver) GetDocument(documentId string) (*register.RegisterDocument, error) {
+func (c InMemoryResolver) GetDocument(documentID string) (*register.RegisterDocument, error) {
+	CountDiscover += 1
```

Comment:
I like the tests this enables :+1: 

Minor point - we probably have none now (though maybe ebus does have - can't remember) - but for multiple routines maybe:
- Increment after `c.lock` locked
- Have vars on the resolver instance. Since `NewInMemoryResolver` returns the struct pointer instead of resolver interface - if the counters were on the struct itself - they should also be accessible, right?